pkg/controller/namespace/namespace_controller.go

```diff
@@ -70,23 +73,24 @@ func NewNamespaceController(
 	// we found in practice though that some auth engines when encountering paths they don't know about may return a 50x.
 	// until we have verbs, we pre-populate resources that do not support list or delete for well-known apis rather than
 	// probing the server once in order to be told no.
-	opCache := operationNotSupportedCache{}
+	opCache := deletion.OperationNotSupportedCache{}
```

Comment:
Make this an interface or something opaque that has an `Ignore(unversioned.GroupVersionResource)` method that you call to below and have deletion check `IsIgnored(gvr)`